windows/src/engine/keyman32/kmprocess.cpp

```diff
@@ -89,6 +89,37 @@ char *getcontext_debug() {
 	return Debug_UnicodeString(_td->app->ContextBufMax(128));
 }
 
+/**
+ *  Process the key stroke using the core processor
+ *
+ * @param   _td   Pointer to KEYMAN64THREADDATA
+ * @return  BOOL  Return TRUE if keystroke event is passed successfully
+ */
+
+static BOOL
+Process_Event_Core(PKEYMAN64THREADDATA _td) {
+  PWSTR contextBuf            = _td->app->ContextBufMax(MAXCONTEXT);
+  km_kbp_context_item *citems = nullptr;
+  ContextItemsFromAppContext(contextBuf, &citems);
+  if (KM_KBP_STATUS_OK !=
+      (km_kbp_status_codes)km_kbp_context_set(km_kbp_state_context(_td->lpActiveKeyboard->lpCoreKeyboardState), citems)) {
```

Comment:
Do we need the `(km_kbp_status_codes)` cast? It's ugly 😢

Reply:
One of the compilers complained about this. I can remove it and see which one it was if it works everywhere I will leave it out.


Why is here an enum `km_kbp_status_codes` defined but we then use a `int` as I return code instead?

Reply:
Yeah I'm not sure why we have `int` instead of `km_kbp_status_codes` as a return type...